nncf/common/tensor_statistics/statistic_point.py

```diff
@@ -71,6 +71,26 @@ def add_statistic_point(self, statistic_point: StatisticPoint) -> None:
 
             self.data[target_node_name].append(statistic_point)
 
+    def remove_statistic_points(self, algorithm_name: str) -> None:
+        """
+        Method to remove statistic point associated with a given algorithm
+        from statistic point container.
+
+        :param algorithm: Algorithm name
+        """
+        data_copy = dict(self.data)
+
+        for target_node_name, statistic_points in data_copy.items():
+            # Reassign every target node name IF it doesn't contain the given algorithm
+            self.data[target_node_name] = [
+                _statistic_point
+                for _statistic_point in statistic_points
+                if algorithm_name not in _statistic_point.algorithm_to_tensor_collectors
+            ]
+
+            if not self.data[target_node_name]:
+                del self.data[target_node_name]
```

Comment:
Hmm, but is it needed to keep empty keys in the dictionary?

Reply:
> Hmm, but is it needed to keep empty keys in the dictionary?

...because in the method where we _add_ the statistical points we **create** the keys that are missing from the `self.data` either way, so might as well just delete the key entirely

Reply:
I think the empty keys are needed in general.
Anyway, making a `dict(self.data)` returns the shallow copy and `statistic_points` remains the same as in `self.data`.
I just wondering if we may do here something more safe to prevent passing through changed `self.data` somehow.

Reply:
> I think the empty keys are not needed in general. Anyway, making a `dict(self.data)` returns the shallow copy and `statistic_points` remains the same as in `self.data`. I just wondering if we may do here something more safe to prevent passing through changed `self.data` somehow.

I made the copy of the dictionary using `dict()` _precisely_ in order to avoid passing through the changed `self.data`
Correct me if I'm wrong, but isn't this sufficient?